### PR TITLE
CPU fixes

### PR DESCRIPTION
**Pull request by valentynbez**
- limit the number of used threads for execution on cluster nodes
- properly deserialise CUDA/CPU
- remove bug that prevented `--help`
- update needed Intel dependencies for CPU

### CODE REVIEW

**Comment by mortonjt** [comment]
Hi @valentynbez thanks for the pull request!  From a first glance, this looks good.  

Out of curiosity, what errors are you running into when you don't install `mkl=2021 mkl_fft`?  We didn't need to install these dependencies (abet need to check the our pre-configuration).

**Comment by valentynbez** [comment]
See issue: https://github.com/facebookresearch/faiss/issues/2890

File: README.md [hunk 1]
```diff
@@ -7,14 +7,14 @@ TM-Vec: template modeling vectors for fast homology detection and alignment: htt
 
 First create a conda environment with python=3.9 installed.  If you are using cpu, use
 
-`conda create -n tmvec faiss-cpu python=3.9 -c pytorch`
+`conda create -n tmvec mkl=2021 mkl_fft faiss-cpu python=3.9 -c pytorch`
```
Comment (mortonjt) [inline README.md:10]:
```suggestion
`conda create -n tmvec faiss-cpu python=3.9 -c pytorch`

If the installation fails, you may need to install mkl via `conda install mkl=2021 mkl_fft `
```

**Comment by mortonjt** [comment]
Great.  I think we can merge as is.  Thanks for your contribution!

**Pull request merged**